finetuner/experiment.py

```diff
@@ -206,40 +200,60 @@ def _create_config_for_run(
         """
         callbacks = kwargs[CALLBACKS] if kwargs.get(CALLBACKS) else []
         callbacks = [
-            {
-                NAME: callback.__class__.__name__,
-                OPTIONS: {
+            config.CallbackConfig(
+                name=callback.__class__.__name__,
+                options={
                     field.name: getattr(callback, field.name)
                     for field in fields(callback)
                 },
-            }
+            )
             for callback in callbacks
         ]
-        return {
-            MODEL: {
-                NAME: model,
-                FREEZE: kwargs.get(FREEZE),
-                OUTPUT_DIM: kwargs.get(OUTPUT_DIM),
-                OPTIONS: kwargs.get(MODEL_OPTIONS) or {},
-                ONNX: kwargs.get(ONNX) or False,
-            },
-            DATA: {
-                TRAIN_DATA: train_data,
-                EVAL_DATA: kwargs.get(EVAL_DATA),
-                NUM_WORKERS: kwargs.get(NUM_WORKERS),
-            },
-            CALLBACKS: callbacks,
-            HYPER_PARAMETERS: {
-                LOSS: kwargs.get(LOSS),
-                OPTIMIZER: kwargs.get(OPTIMIZER),
-                OPTIMIZER_OPTIONS: kwargs.get(OPTIMIZER_OPTIONS) or {},
-                MINER: kwargs.get(MINER),
-                MINER_OPTIONS: kwargs.get(MINER_OPTIONS) or {},
-                BATCH_SIZE: kwargs.get(BATCH_SIZE),
-                LEARNING_RATE: kwargs.get(LEARNING_RATE),
-                EPOCHS: kwargs.get(EPOCHS),
-                SCHEDULER_STEP: kwargs.get(SCHEDULER_STEP),
-            },
-            EXPERIMENT_NAME: experiment_name,
-            RUN_NAME: run_name,
-        }
+        model = config.ModelConfig(
+            name=model,
+            output_dim=kwargs.get(OUTPUT_DIM),
+        )
+        if kwargs.get(FREEZE):
```

Comment:
no need to do all these ifs. keep the old structure so that we dont fall into any new traps
```
run_config = config.RunConfig(
    model=config.ModelConfig(
        name=model,
        output_dim=kwargs.get(OUTPUT_DIM),
        freeze = kwargs.get(FREEZE),
        options = kwargs.get(MODEL_OPTIONS) or {},
        to_onnx = kwargs.get(ONNX) or False,
    ),
    data=config.DataConfig(
        train_data=train_data,
        eval_data=kwargs.get(EVAL_DATA),
        num_workers = kwargs.get(NUM_WORKERS),
    ),
    ...
)
return run_config.dict()
```

Reply:
to_onnx = kwargs.get(ONNX) or False -> `kwargs.get(ONNX, False)`

Reply:
The previous structure led to errors, for example, in the event that `freeze` had not been provided as an argument, `freeze = kwargs.get(FREEZE),` would attempt to set `freeze` to None, causing validation to fail

Reply:
In cases such as `options = kwargs.get(MODEL_OPTIONS) or {},` I also chose to replace with ifs because, while an error would not occur if `MODEL_OPTIONS` is None, this still relies on the alternative value, in this case `{}` being the same as the default value as the field in the config class. Currently `{}` is the default value but I figured this method would reduce the opportunity to make errors in the future

Reply:
yes you are right about the default value, let me think

Reply:
ok you are right, leave it with ifs. We will refactor this anyway, when we try to automatically generate the fit signature from the run config